src/categoriesdialog.cpp

```diff
@@ -234,21 +267,124 @@ void CategoriesDialog::addCategory_clicked()
   ui->categoriesTable->setItem(row, 0,
                                new QTableWidgetItem(QString::number(++m_HighestID)));
   ui->categoriesTable->setItem(row, 1, new QTableWidgetItem("new"));
-  ui->categoriesTable->setItem(row, 2, new QTableWidgetItem(""));
-  ui->categoriesTable->setItem(row, 3, new QTableWidgetItem("0"));
+  ui->categoriesTable->setItem(row, 2, new QTableWidgetItem("0"));
+  ui->categoriesTable->setItem(row, 3, new QTableWidgetItem(""));
 }
 
 void CategoriesDialog::removeCategory_clicked()
 {
   ui->categoriesTable->removeRow(m_ContextRow);
 }
 
+void CategoriesDialog::removeNexusMap_clicked()
+{
+  ui->categoriesTable->item(m_ContextRow, 3)->setData(Qt::UserRole, QVariantList());
+  ui->categoriesTable->item(m_ContextRow, 3)->setData(Qt::DisplayRole, QString());
+  // ui->categoriesTable->update();
+}
+
+void CategoriesDialog::nexusRefresh_clicked()
+{
+  NexusInterface& nexus = NexusInterface::instance();
+  nexus.setPluginContainer(m_PluginContainer);
```

Comment:
I don't really understand what this does here and this is the only place where the plugin container is used in the dialog, so this single line mean the plugin container needs to be pass all the way down here for not much.

Reply:
It was to avoid having to take multiple steps to route the signal to the Nexus API and get useable data back to the Dialog.

But I've done so now anyway.